server/aesdsocket.c

```diff
@@ -0,0 +1,338 @@
+    {
+        syslog(LOG_ERR,"bind failed");
+        return -1;
+    }
+
+    // turn socket into a passive socket to accept connections, 10 connections in queue
+    if (listen(sockfd, 10) == -1) 
+    {
+        syslog(LOG_ERR,"Not listening"); 
+        return -1;
+    }
+
+    freeaddrinfo(res);
+    res = NULL;
+
+
+    //printf("Socket bound to port %s\n", PORT_NUM);
+
+
+    // Skeleton gotten and adapted from Bing Chat gpt 4 "how to create a daemon process!"
+    if(runAsDaemon)
+    {
+        pid_t pid = fork();
+
+        if(pid < 0)
+        {
+            syslog(LOG_ERR,"Fork fail");
+        }
+
+        if(pid > 0)
+        {
+            syslog(LOG_USER,"Exit parent");
+            exit(0);
+        }
+
+        if(setsid() == -1)
+        {
+            syslog(LOG_ERR,"failed to set sid for child remaining");
+        }
+
+        // close all paths to run in background as a daemon
+        close(STDIN_FILENO);
+        close(STDOUT_FILENO);
+        close(STDERR_FILENO);
+
+        
+        if(chdir("/") == -1)
+        {
+            syslog(LOG_ERR,"failed to chdir to /");
+        }
+        
+        umask(0);
+
+    }
+    
+
+    struct sockaddr_storage client_addr;
+    socklen_t addr_size = sizeof(client_addr);
+    char buffer[BUFFER_SIZE];
+    char sendBuffer[BUFFER_SIZE];
+
+    while (1) 
+    {
+        
+        openConSock = accept(sockfd, (struct sockaddr*)&client_addr, &addr_size);
+
+        if(openConSock == -1)
+        {
+            syslog(LOG_ERR,"connection accept failed");
+        }
+        
+        // Implentation is a combination of both cases of either 4 or 6!
+        //https://stackoverflow.com/questions/3060950/how-to-get-ip-address-from-sock-structure-in-c
+
+        char ipAccepted[INET6_ADDRSTRLEN]; // handle both Inet6 and 4
+        void *addrPtr;
+
+        // if INET 4 
+        if(client_addr.ss_family == AF_INET)
+        {
+            // then save the address
+            struct sockaddr_in *tempSock = (struct sockaddr_in *)&client_addr;
+            addrPtr = &(tempSock->sin_addr);
+        }
+        else
+        {
+            // save the INET6 address
+            struct sockaddr_in6 *tempSock = (struct sockaddr_in6 *)&client_addr;
+            addrPtr = &(tempSock->sin6_addr);
+        }
+
+        // translate the pointer to a ip using the INET_NTOP function
+        inet_ntop(client_addr.ss_family, addrPtr, ipAccepted, sizeof(ipAccepted));
+
+        syslog(LOG_INFO,"Accepted connection from %s",ipAccepted);
+
+
+        // open the writing file and create it if it doesn't exist!
+        writefp = open(DATA_PATH,   O_RDWR | O_CREAT |  O_APPEND, 0664);
+
+        if (writefp == -1 ) 
+        {
+             syslog(LOG_ERR,"Open write file failed");
+            return -1;
+        }
+
+        int bytes_received = 1;
+        char lastChar = ' ';
+
+        // While buffer recieved end is not '\n'
+        while (lastChar != '\n') 
```

Comment:
@Nicholas-Buckley77 what about the case where the `\n` is inside the buffer rather than the last char?  Is there a guarantee that `recv` will send exactly the number of bytes in the packet?